### PR TITLE
feat: implement the machine categories UI

### PR DESCRIPTION
**Pull request by Unix4ever**
It is now possible to filter the machine by the following categories in the UI:
- Manually joined machines.
- Machines provisioned by the infra providers.
- Machines PXE booted through the metal provider.
- Machines discovered by the metal provider, but not yet accepted by the system.

Machine acceptance UI also has two new modal windows for accepting and rejecting the machines.
Also update controllers to add labels to the not accepted infra machines.

Fixes: https://github.com/siderolabs/omni/issues/773
Fixes: https://github.com/siderolabs/omni/issues/766

### CODE REVIEW

**Review by smira — APPROVED** [review]
visually looks :cool:

**Comment by TimJones** [comment]
Looks great, layout is much, but I'm not sure if I understand the difference between "Manual" and "PXE Booted"...

File: internal/backend/runtime/omni/controllers/omni/infra_machine.go [hunk 1]
```diff
@@ -164,13 +165,18 @@ func (h *infraMachineControllerHelper) finalizerRemovalExtraOutput(ctx context.C
 		return err
 	}
 
+	_, err = helpers.HandleInput[*omni.InfraMachineConfig](ctx, r, InfraMachineControllerName, link)
```
Comment (utkuozdemir) [inline internal/backend/runtime/omni/controllers/omni/infra_machine.go:168]: 👍

**Review by utkuozdemir — APPROVED** [review]

File: internal/version/data/tag [hunk 2]
```diff
@@ -1 +1 @@
-v0.45.0-beta.0
+v0.45.0-beta.0
```
[the remark this answers is not in the document]
Comment (Unix4ever) [reply internal/version/data/tag:1]: 🤷

**Pull request merged**